src/librustc_resolve/resolve_imports.rs

```diff
@@ -819,28 +819,48 @@ impl<'a, 'b:'a, 'c: 'b> ImportResolver<'a, 'b, 'c> {
         }
     }
 
-    fn throw_unresolved_import_error(&self, error_vec: Vec<(Span, String, String)>,
-                                     span: Option<MultiSpan>) {
+    fn throw_unresolved_import_error(
+        &self,
+        error_vec: Vec<(Span, String, String, Option<String>)>,
+        span: Option<MultiSpan>,
+    ) {
         let max_span_label_msg_count = 10;  // upper limit on number of span_label message.
-        let (span,msg) = match error_vec.is_empty() {
-            true => (span.unwrap(), "unresolved import".to_string()),
+        let (span, msg, note) = match error_vec.is_empty() {
+            true => (span.unwrap(), "unresolved import".to_string(), None),
             false => {
-                let span = MultiSpan::from_spans(error_vec.clone().into_iter()
-                                    .map(|elem: (Span, String, String)| { elem.0 }
-                                    ).collect());
+                let span = MultiSpan::from_spans(
+                    error_vec.clone().into_iter()
+                    .map(|elem: (Span, String, String, Option<String>)| elem.0)
+                    .collect()
+                );
+
+                let note = error_vec.clone().into_iter()
+                    .filter(|elem: &(Span, String, String, Option<String>)| elem.3.is_some())
+                    .map(|elem: (Span, String, String, Option<String>)| elem.3.unwrap())
+                    .collect::<Vec<String>>()
+                    .pop();
```

Comment:
This isn't a particularly nice way to get the note, but outside of a larger refactoring of the error reporting in this module (which seemed out of scope for this PR), this was the cleanest way I could figure to get access to it.

Reply:
Why are you `collect`ing into a vec and then immediately `pop`ing off it? Wouldn't [Iterator::last](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.last) do the trick here?

Reply:
also, I think you can use [Iterator::filter_map]() to make the two separate stages of `filter` and `map` into one nice stage

Reply:
Fixed!